awx/main/signals.py

```diff
@@ -72,40 +72,65 @@ def get_current_user_or_none():
     return u
 
 
-def emit_event_detail(serializer, relation, **kwargs):
+def emit_event_detail(cls, relation, **kwargs):
     instance = kwargs['instance']
     created = kwargs['created']
     if created:
-        event_serializer = serializer(instance)
+        url = ''
+        if isinstance(instance, JobEvent):
+            url = '/api/v2/job_events/{}'.format(instance.id)
+        if isinstance(instance, AdHocCommandEvent):
+            url = '/api/v2/ad_hoc_command_events/{}'.format(instance.id)
+        group = camelcase_to_underscore(cls.__name__) + 's'
+        timestamp = instance.created.isoformat()
         consumers.emit_channel_notification(
-            '-'.join([event_serializer.get_group_name(instance), str(getattr(instance, relation))]),
-            event_serializer.data
+            '-'.join([group, str(getattr(instance, relation))]),
+            {
+                'id': instance.id,
+                relation.replace('_id', ''): getattr(instance, relation),
+                'created': timestamp,
+                'modified': timestamp,
+                'group_name': group,
+                'url': url,
+                'stdout': instance.stdout,
+                'counter': instance.counter,
+                'uuid': instance.uuid,
+                'start_line': instance.start_line,
+                'end_line': instance.end_line,
+                'event': instance.event,
+                'failed': instance.failed,
+                'changed': instance.changed,
+                'event_level': getattr(instance, 'event_level', ''),
+                'play': getattr(instance, 'play', ''),
+                'role': getattr(instance, 'role', ''),
+                'task': getattr(instance, 'task', ''),
+            }
```

Comment:
needed:

- `event.parent_uuid`
- `event.event_data` 
- `event.job`
- `event.project_update`
- `event.ad_hoc_command`
- `event.event_level`
- `event.reason`
- `event.host`
- `event.status`

Reply:
`job` `project_update` and `ad_hoc_command` should be handled by this line: https://github.com/ansible/awx/pull/5618/files/68942365cf3a734e2b813bf7cafeb51042c4a4b5#diff-840aee1d09c364211891211e2a4cef7eR90

Reply:
These don't look like fields on job events to me at `/api/v2/job_events/`:

```
event.reason
event.status
```